lib/internal/test_runner/test.js

```diff
@@ -552,23 +552,20 @@ class Suite extends Test {
     try {
       const context = { signal: this.signal };
       this.buildSuite = this.runInAsyncScope(this.fn, context, [context]);
+      if (isPromise(this.buildSuite)) {
+        this.buildSuite = this.buildSuite.catch((err) => {
+          this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
+        });
+      }
     } catch (err) {
       this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
     }
     this.fn = () => {};
     this.buildPhaseFinished = true;
   }
 
-  start() {
-    return this.run();
-  }
-
   async run() {
-    try {
-      await this.buildSuite;
-    } catch (err) {
-      this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
-    }
+    await this.buildSuite;
```

Comment:
caching here is too late in cases where `run` is queued

Reply:
Any chance we could trigger that specific case with a test? My attempts have failed so far, so if it's not possible, so be it, but it seems like something our tests should cover.

Reply:
> yes, this test will cover it (run this file before and after this fix and you will notice the difference): https://github.com/nodejs/node/blob/387c21e1e65df4af71995eb72f5467d0540778c1/test/message/test_runner_hooks.js

If it's fixed with this PR, it makes sense to land the test here. Or am I missing something?

Reply:
What if the other PR never lands, or has to be reverted for whatever reason? If it's not possible (or unreasonably hard to do), that's fair enough, it's not a blocking concern anyway, but it's much better if it can happen in the same PR.

Reply:
Ok then :) Il add some tests

Reply:
added